### PR TITLE
[DEVELOPER-3605] Export process should fail if there are pages missing

### PR DESCRIPTION
**Pull request by robpblake**
This PR updates the export process to fail if there are missing pages. Previously the behaviour was to simply emit a warning if there were pages missing from the export, however these were going un-noticed and the site was being published with pages missing.

I have also included a mechanism to override this default failure behaviour if you absolutely need to. By setting the `drupal.export.fail_on_missing` environment variable to `false`, you can disable the failure behaviour. This allows you to push an export in times when bad things are happening and you just need to get something up. 

I've added the `drupal.export.fail_on_missing` variable to the `export` container definitions in docker-compose.yml for each environment. The variable is currently set to nothing meaning that the failure behaviour is enabled by default. Should we ever need to disable the failure behaviour, then we can simply set the value of the `drupal.export.fail_on_missing` variable in the Jenkins job configuration for the export.

### CODE REVIEW

**Comment by robpblake** [comment]
@LightGuard This updates the export process to fail if there are pages missing. Appreciate if you could give the small code changes the once over and also test the export locally to make sure it works for you as well.

cheers,

Rob

File: _docker/lib/export/export_inspector.rb [hunk 1]
```diff
@@ -21,6 +26,21 @@ def determine_expected_filesystem_path(url)
   end
 
   #
+  # Determines if the process should fail if content is missing from the export
+  #
+  def should_fail_on_missing_content
+    fail_on_missing = ENV['drupal.export.fail_on_missing']
+    fail = true
+
+    if !fail_on_missing.nil? && !fail_on_missing.empty?
```
Comment (LightGuard) [inline _docker/lib/export/export_inspector.rb]: Wouldn't

```
unless fail_on_missing.nil && fail_on_missing.empty?
```
read a bit easier?

File: _docker/lib/export/export_inspector.rb [hunk 2]
```diff
@@ -21,6 +26,21 @@ def determine_expected_filesystem_path(url)
...
+      if fail_on_missing.to_s == 'false'
```
Comment (LightGuard) [inline _docker/lib/export/export_inspector.rb]: In ruby you'd typically use a match for this:
```
fail = false if fail_on_missing =~ /false/
```

File: _docker/lib/export/export_inspector.rb (hunk 1 above)
Comment (robpblake) [inline _docker/lib/export/export_inspector.rb]: Yep.

File: _docker/lib/export/export_inspector.rb (hunk 2 above)
Comment (robpblake) [inline _docker/lib/export/export_inspector.rb]: Thanks.

**Comment by robpblake** [comment]
retest this please

**Comment by robpblake** [comment]
This is now failing because there are pages missing from the export. So at least we know the code works I guess...

**Comment by robpblake** [comment]
@LightGuard I've made the changes you proposed. This is now failing as there are pages missing from the export, so I need to address that first.

**Comment by robpblake** [comment]
retest this please

**Comment by robpblake** [comment]
@LightGuard I've updated the DB dump on stumpjumper to be the latest production backup and this now seems to be working fine. Ready for review and merge when you're ready.

cheers,

Rob

**Pull request merged**